lib/smart_proxy_openscap/foreman_forwarder.rb

```diff
@@ -3,23 +3,19 @@ class ForemanForwarder < Proxy::HttpRequest::ForemanRequest
     include ::Proxy::Log
 
     def post_arf_report(cname, policy_id, date, data)
-        parser = Proxy::OpenSCAP::Parse.new(data)
       begin
         foreman_api_path = upload_path(cname, policy_id, date)
-        response = send_request(foreman_api_path, parser.as_json)
+        json = Proxy::OpenSCAP::Parse.new(cname, policy_id, date).as_json(data)
+        raise OpenSCAP::OpenSCAPError, "Failed to parse report" if json.empty?
```

Comment:
the result in `json` can also be instance of Tempfile in case some exception was raised, the method should perhaps be explicit and return `nil` in case something went wrong, on this line only nil should be checked